airflow/docker_image.go

```diff
@@ -48,23 +49,52 @@ func DockerImageInit(image string) *DockerImage {
 	return &DockerImage{imageName: image}
 }
 
-func (d *DockerImage) Build(dockerfile, buildSecretString string, buildConfig airflowTypes.ImageBuildConfig) error {
+func shouldAddPullFlag(dockerfilePath string) (bool, error) {
+	file, err := os.Open(dockerfilePath)
+	if err != nil {
+		return false, fmt.Errorf("opening file: %w", err)
+	}
+	defer file.Close()
+
+	// scan the Dockerfile to check if any FROM instructions reference an image other than astro runtime
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "FROM ") && !strings.HasPrefix(line, fmt.Sprintf("FROM %s:", FullAstroRuntimeImageName)) {
```

Comment:
Could we just rely on `astro-runtime` so we are not tied to a specific image registry?

Reply:
resolved offline